### PR TITLE
[mobile] Fix Conv2d string padding handling in XNNPACK optimization

### LINKED ISSUE #65490: optimize_for_mobile() assert fail: missing op for prepacked::conv2d_clamp_prepack
## 🐛 Bug

When calling torch.utils.mobile_optimizer.optimize_for_mobile, an internal assert fails:

```
0INTERNAL ASSERT FAILED at "../torch/csrc/jit/ir/alias_analysis.cpp":532, please report a bug to PyTorch. We don't have an op for prepacked::conv2d_clamp_prepack but it isn't a special case.  Argument types: Tensor, Tensor, int[], str, int[], int, NoneType, NoneType,
```

## To Reproduce

Created convolutional model (gapnet plus linear -> sigmoid head).  Resulting structure as output by pytorch:

```
FullGuidanceModel(
  (plan_model): GuidancePlanModel(
    (backbone): GapSmall(
      (model): Sequential(
        (0): TensorflowSamePadForConv2D()
        (1): Conv2d(1, 16, kernel_size=(3, 3), stride=(2, 2), padding=valid)
        (2): ELU(alpha=1.0)
        (3): TensorflowSamePadForConv2D()
        (4): Conv2d(16, 32, kernel_size=(3, 3), stride=(2, 2), padding=valid)
        (5): ELU(alpha=1.0)
        (6): TensorflowSamePadForConv2D()
        (7): Conv2d(32, 64, kernel_size=(3, 3), stride=(2, 2), padding=valid)
        (8): ELU(alpha=1.0)
        (9): TensorflowSamePadForConv2D()
        (10): Conv2d(64, 128, kernel_size=(3, 3), stride=(2, 2), padding=valid)
        (11): ELU(alpha=1.0)
        (12): TensorflowSamePadForConv2D()
        (13): Conv2d(128, 128, kernel_size=(3, 3), stride=(2, 2), padding=valid)
        (14): ELU(alpha=1.0)
      )
    )
    (logits_dense): Linear(in_features=128, out_features=1, bias=True)
  )
  (moving_avg_layer): MovingAverageAcrossBatchWithMemory()
)
```
Note: TensorflowSamePadForConv2D modules just apply a pre-calculated F.pad() operation to their input and return.  GapSmall's output gets averaged over spatial dimensions.

## Expected behavior

scripted module compiles normally.

## Environment

 - PyTorch Version (e.g., 1.0): 1.9.1+cu102
 - OS (e.g., Linux): Ubuntu 20.04 (WSL)
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8.7
 - CUDA/cuDNN version: n/a
 - GPU models and configuration: n/a
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by AkiSakurai**
Prevents rewriting Conv2d operations that use string-based padding modes ('same') to prepacked ops. This fixes internal assertion failures in alias analysis when handling string padding values.

Changes:
- Added padding mode filter to Conv2d rewrite patterns and applied it to regular/transpose operations
- Added test coverage for Conv2d with 'same' padding mode

Fixes #65490

cc @EikanWang @jgong5 @wenzhe-nrv @sanchitintel

### CODE REVIEW

**Review by davidberard98 — COMMENTED** [review]
generally lgtm, will give @digantdesai a day to review first though

File: torch/csrc/jit/passes/xnnpack_rewrite.cpp [hunk 1]
```diff
@@ -118,6 +118,13 @@ void insertPrePackedConv2dOp(std::shared_ptr<Graph>& graph) {
   // Replace _convolution with conv2d
   graph_rewrite_helper::replaceConvolutionWithAtenConv(graph);
 
+  auto filter_padding =
+      [](const Match& match,
+         const std::unordered_map<std::string, Value*>& vmap) {
+        auto* node = match.nodes_map.at(vmap.at("res")->node());
+        return node->schema().overload_name() != "padding";
```
Comment (davidberard98) [inline torch/csrc/jit/passes/xnnpack_rewrite.cpp:125]: same comment as https://github.com/pytorch/pytorch/pull/140221#discussion_r1844233713

**Review by davidberard98 — CHANGES_REQUESTED** [review]
